homeassistant/components/switch/modbus.py

```diff
@@ -77,3 +122,82 @@ def update(self):
                 'No response from modbus slave %s coil %s',
                 self._slave,
                 self._coil)
+
+
+class ModbusRegisterSwitch(ModbusCoilSwitch):
+    """Representation of a Modbus register switch."""
+
+    # pylint: disable=super-init-not-called
+    def __init__(self, name, slave, register, command_on,
+                 command_off, verify_state, verify_register,
+                 register_type, state_on, state_off):
+        """Initialize the register switch."""
+        self._name = name
+        self._slave = int(slave) if slave else None
```

Comment:
Config validation has already made sure this is a positive integer if it's not `None`. So just assign the argument to the instance attribute.